### PR TITLE
Allowing for containerd to use zfs snapshotter by default

### PR DESCRIPTION
**Pull request by rvs**
Here's a tiny tweak to unblock further experiments with ZFS.

@cshari-zededa this is what I suggested as a way to handle similar things in vaultmgr

### CODE REVIEW

File: pkg/pillar/containerd/containerd.go [hunk 1]
```diff
@@ -90,6 +92,11 @@ func InitContainerdClient() error {
 	}
 	if contentStore == nil {
 		contentStore = CtrdClient.ContentStore()
+		// see if we need to fine-tune default snapshotter based on what flavor of storage persist partition is
+		persistType, err := ioutil.ReadFile(eveStorageTypeFile)
+		if err == nil && string(persistType) == "zfs" {
```
Comment (deitch) [inline pkg/pillar/containerd/containerd.go:97]: Might want to make this a `switch`/`case` statement; I figure we will have other types in the future? Maybe not?

Comment (deitch) [reply pkg/pillar/containerd/containerd.go:97]: Also, what do we do if the file does not exist? We should fall back to the original `defaultSnapshotter`

**Comment by deitch** [comment]
Here is what I don't get. All this does is allows us to pass a different string to containerd to use a snapshotter named `"zfs"`. That doesn't do a lot of good if such a named snapshotter doesn't exist. Is this just the plumbing so we can add it in the (likely near) future?

File: pkg/pillar/containerd/containerd.go (hunk 1 above)
Comment (eriknordmark) [inline pkg/pillar/containerd/containerd.go:97]: If we expect folks to manually edit this file you should probably strip whitespace like the CR at the end to make the comparison work.

File: pkg/storage-init/storage-init.sh [hunk 2]
```diff
@@ -148,11 +153,13 @@ if P3=$(findfs PARTLABEL=P3) && [ -n "$P3" ]; then
         fi
     fi
 
+    # deposit fs type into /run
+    echo "$P3_FS_TYPE" > /run/eve.persist_type
```
Comment (eriknordmark) [inline pkg/storage-init/storage-init.sh:157]: Actually this adds a CR at the end thus AFAICT the readfile+comparison with "zfs" will always fail.

File: pkg/pillar/containerd/containerd.go (hunk 1 above)
Comment (rvs) [reply pkg/pillar/containerd/containerd.go:97]: > Also, what do we do if the file does not exist? We should fall back to the original defaultSnapshotter

the original one is always "overlayfs" -- that didn't change -- so think about this snippet of code as an opportunistic override for now

Comment (rvs) [reply pkg/pillar/containerd/containerd.go:97]: Great catch @eriknordmark -- will fix this

File: pkg/storage-init/storage-init.sh (hunk 2 above)
Comment (rvs) [reply pkg/storage-init/storage-init.sh:157]: Yup -- as was reported by @cshari-zededa as well

**Comment by rvs** [comment]
> Here is what I don't get. All this does is allows us to pass a different string to containerd to use a snapshotter named "zfs". That doesn't do a lot of good if such a named snapshotter doesn't exist. Is this just the plumbing so we can add it in the (likely near) future?

@deitch not sure what you're missing here, but zfs snapshotter is bundled by default with linuxkit containerd. So all that is required to "enable it" is having the dataset mounted in the /var/lib/containerd/.... (and obviously requesting it -- which is what this PR is about)

**Comment by deitch** [comment]
> zfs snapshotter is bundled by default with linuxkit containerd

Oh right..... 🤦🏻‍♂️

**Pull request merged**